tern/analyze/docker/dockerfile.py

```diff
@@ -118,6 +125,27 @@ def parse_from_image(dfobj):
     return image_list
 
 
+def expand_from_images(dfobj):
+    '''Replace all parent_image values with their full tag@sha:digest value.
+    Update the structure dictionary with the same information.
+    dfobj: the Dockerfile object created using get_dockerfile_obj'''
+    # update parent_images
+    images = parse_from_image(dfobj)
+    for i, img in enumerate(images):
+        if general.check_image_string(img['name'] + tag_separator +
+                                      img['tag']):
+            dfobj.parent_images[i] = container.get_image_digest(
+                container.get_image(img['name'] + tag_separator + img['tag']))
+    # update structure
+    for i, command_dict in enumerate(dfobj.structure):
+        if command_dict['instruction'] == 'FROM':
+            dfobj.structure[i]['content'] = command_dict['instruction'] + \
+                ' ' + container.get_image_digest(container.get_image(
+                    command_dict['value'])) + '\n'
```

Comment:
Why are you trying to `get_image` again?

Reply:
The value of the digest needs to be assigned twice. But I can assign it to a variable and assign it that way if you prefer.

Reply:
The value of the digest needs to be assigned twice and therefore needs the image object twice as input. But I can assign the image object to a variable and input it to get_image_digest that way if you prefer.

Reply:
Additionally, this is the first time `get_image` is called in this loop -- so not sure what you mean by "again" unless you're talking about line 145-146? Or are you saying again from the loop above?

Reply:
Why not use what is stored in `dfobj.parent_images`? Or if you are worried about losing association, how about making a dictionary with `{'image:tag': 'image@digest_type:digest'}` in the first loop and using it in the second loop?

Reply:
I thought about that but since we are iterating through a structure with many instructions, the count of the second loop will be extremely difficult to keep up with the one or two images in the images dictionary.